redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -0,0 +1,730 @@
+                        }
+
+                        // Prepare MessageArgs as per defined in the
+                        // MessageRegistries
+                        std::vector<std::string> messageArgs{*msgPropVal};
+
+                        // Fill the "msgPropVal" as reason
+                        std::string message = msgReg->message;
+                        int i = 0;
+                        for (const std::string& messageArg : messageArgs)
+                        {
+                            std::string argIndex = "%" + std::to_string(++i);
+                            size_t argPos = message.find(argIndex);
+                            if (argPos != std::string::npos)
+                            {
+                                message.replace(argPos, argIndex.length(),
+                                                messageArg);
+                            }
+                        }
+                        // Severity will be added based on the event
+                        // object property
+                        condition["Message"] = message;
+                        condition["MessageArgs"] = messageArgs;
+                        condition["MessageId"] =
+                            "OpenBMC.0.2.HardwareIsolationReason";
+                    }
+                    else if (property.first == "Severity")
+                    {
+                        const std::string* severity =
+                            std::get_if<std::string>(&property.second);
+                        if (severity == nullptr)
+                        {
+                            BMCWEB_LOG_ERROR
+                                << "Failed to get the Severity "
+                                << "from object: " << hwStatusEventObj;
+                            messages::internalError(aResp->res);
+                            return;
+                        }
+
+                        // we have only one condition
+                        nlohmann::json::json_pointer severityPropPath =
+                            "/Status/Conditions/0/Severity"_json_pointer;
+                        if (!setSeverity(aResp, hwStatusEventObj,
+                                         severityPropPath, *severity))
+                        {
+                            // Failed to set the severity
+                            return;
+                        }
+                    }
+                }
+                },
+                objType[0].first, hwStatusEventObj,
+                "org.freedesktop.DBus.Properties", "GetAll", "");
+            },
+            "xyz.openbmc_project.ObjectMapper",
+            "/xyz/openbmc_project/object_mapper",
+            "xyz.openbmc_project.ObjectMapper", "GetObject", hwStatusEventObj,
+            std::array<const char*, 1>{"xyz.openbmc_project.Logging.Event"});
+        },
+        "xyz.openbmc_project.ObjectMapper", resourceObjPath.str + "/event_log",
+        "org.freedesktop.DBus.Properties", "Get",
```

Comment:
sdbusplus::asio::getProperty

Reply:
Thank you for your valid comments. We have created the JIRA ticket https://jsw.ibm.com/browse/PFEBMC-620 to address this comment.
The scope of this task was to cherry pick the working HWIsolation commits from 1030. And as the number of commits are more, it is difficult to refactor them to align with upstream as part of this PR. So we will address all them as a separate story.